CONTENTdm.js

```diff
@@ -0,0 +1,577 @@
+  else if ( url.match( "/cdm/landingpage" ) || !url.substring( url.indexOf( '/cdm/' ) + 5 ) ) return "webpage";
+  else if ( url.match( "/cdm/fullbrowser" ) ) { // Pages without item metadata
+	originatingItem = ZU.xpath( doc, '//link[@rel="canonical"]' );
+	if ( originatingItem && originatingItem[ 0 ].href ) return "document"; // Slow to scrape again for item type
+	else return "";
+  }
+  return ( function() {
+	var itemType = fetchXPathText( doc, '//*[@id="metadata_type"]' );
+	if ( !itemType ) itemType = fetchXPathText( doc, '//*[@id="metadata_object_type"]' );
+	switch ( itemType ) {
+	  case "Image":
+	  case "Photograph":
+	  case "Engraved portrait":
+	  case "Negative":
+		return 'artwork';
+	  case "Text":
+		return 'manuscript';
+	  default:
+		return 'document';
+	}
+  }() );
+}
+
+function fetchXPathText( doc, xpath ) { // Removes excess whitespace & breaks, respaces items overtrimmed by trimInternal.
+  if ( ZU.xpathText( doc, xpath ) ) {
+	return ZU.trimInternal( ZU.xpathText( doc, xpath ).replace( /^\s+|\s+$|\;$|\,$/g, '' ).replace( /([a-z])([A-Z])/g, '$1\. $2' ) );
+  } else return;
+}
+
+function fetchXPathValue( doc, xpath ) {
+  if ( ZU.xpath( doc, xpath )[ 0 ] ) return ZU.xpath( doc, xpath )[ 0 ].value;
+  else return;
+}
+
+function fetchXmlUrl( doc, url ) { // Fetches xml urls if using OCLC server (as constructor)
+  var serverID = new RegExp( "/collection\/p([A-Z0-9]+)coll", "i" );
+  var serverID2 = new RegExp( "//cdm([0-9]+).contentdm.oclc.org" );
+  var collectionID = new RegExp( "/collection\/([A-Z0-9]+)\/", "i" );
+  var objectID = new RegExp( "/id\/([A-Z0-9]+)", "i" );
+  var itemID = new RegExp( "/show\/([A-Z0-9]+)", "i" );
+  var objectUrlHttp = fetchXPathValue( doc, '//input[@id="cdm_objectRefUrl"]' );
+  var itemUrlHttp = fetchXPathValue( doc, '//input[@id="cdm_newRefUrl"]' );
+  var server;
+  var xmlQueryUrl;
+  if ( objectUrlHttp ) url = objectUrlHttp; // Use object page as primary even if item page is present
+  else if ( itemUrlHttp ) {
+	url = itemUrlHttp;
+	itemUrlHttp = "";
+  } // Use item page as primary if no object page is present
+  if ( url.match( serverID ) ) server = url.match( serverID )[ 1 ];
+  else if ( url.match( serverID2 ) ) server = url.match( serverID2 )[ 1 ];
+  if ( server && url.match( collectionID ) && url.match( objectID ) ) {
+	xmlQueryUrl = "https://server" + server + ".contentdm.oclc.org/dmwebservices/index.php?q=dmGetItemInfo/" + url.match( collectionID )[ 1 ] + "/";
+	this.object = xmlQueryUrl + url.match( objectID )[ 1 ] + "/show/xml";
+  } else this.object = "";
+  if ( itemUrlHttp ) this.item = xmlQueryUrl + itemUrlHttp.match( objectID )[ 1 ] + "/show/xml";
+  else if ( this.object && url.match( itemID ) ) this.item = xmlQueryUrl + url.match( itemID )[ 1 ] + "/show/xml";
+  else this.item = "";
+}
+
+function scrape( doc, url ) {
+  var item = new Zotero.Item( detectWeb( doc, url ) );
+  url = doc.documentURI;
+  var xmlUrl = new fetchXmlUrl( doc, url );
```

Comment:
If you want to return more than one parameter, return an object. Don't do this, it's unconventional and confusing.